File: arry/217. 存在重复元素/go/Contains Duplicate.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	data := []int{1, 2, 3, 1}
	fmt.Println(containsDuplicate(data))
}

func containsDuplicate(nums []int) bool {
	sort.Ints(nums)
	for i := 1; i < len(nums); i++ {
		if nums[i] == nums[i-1] {
			return true
		}
	}
	return false
}

File: leetcode-arry/217. 存在重复元素/python/Contains Duplicate.py
def containsDuplicate(nums)->bool:
    ret = False
    # 解法一 直接利用集合去重的特性比对数组长度
    # ret = False
    # length = len(nums)
    # if length != len(set(nums)):
    #     ret = True

    # 解法二 排序比较相邻的两个元素是否相等
    nums.sort()
    for i in range(len(nums)-1):
        if nums[i]== nums[i+1]:
            ret = True
    return ret
